pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -117,6 +116,16 @@ func splitHostIntPort(s string) (string, int, error) {
 	return host, portInt, err
 }
 
+func validatePercentageOfNodesToScore(path *field.Path, percentageOfNodesToScore *int32) []error {
+	var errs []error
```

Comment:
why define a slice of error instead of a single error?

Reply:
> why define a slice of error instead of a single error?

For consistency and generalicity (extensibility), all other validate funcs return `[]error`.

Reply:
> I recalled it was not needed anymore? Accidentally added back?
We still need it even with the change of percentageToNodesToScore to pointer. `make update` won't generate the function for us.

Reply:
> For consistency and generality (extensibility)

Doesn't quite make sense to me as we'd _always_ parse one field here. We should use []errors only when necessary.

Reply:
> > For consistency and generality (extensibility)
> 
> Doesn't quite make sense to me as we'd _always_ parse one field here. We should use []errors only when necessary.

Changed `[]error` to `error`.

Reply:
@Huang-Wei we do this all the time in validation code.

It's easy to do `errs = append(errs, validateSubField()...)`. And you save an extra `if`. But I won't block on this.